packages/analytics-browser/src/plugins/form-interaction-tracking.ts

```diff
@@ -8,7 +8,31 @@ import {
 } from '../constants';
 import { BrowserConfig } from '../config';
 
+interface EventListener {
+  element: Element;
+  type: 'change' | 'submit';
+  handler: () => void;
+}
+
 export const formInteractionTracking = (): EnrichmentPlugin => {
+  let observer: MutationObserver | undefined;
+  let eventListeners: EventListener[] = [];
+  const addEventListener = (element: Element, type: 'change' | 'submit', handler: () => void) => {
```

Comment:
nit: I find it confusing to see a function named `addEventListener`, my brain immediately jumps to "is this calling `window.addEventListener` implicitly? Same with remove. Nbd